### ISSUE TITLE
Allow branching from a remote instance

### ISSUE THREAD

**Issue by scotttrinh** [edgedb/edgedb-cli#1342]
Right now, the `branch create` command only operates on the connected instance. It would be useful to be able to specify different instances for the source and target so you can, for instance, copy just the schema of your production database into CI without having to manually create a schema-only branch and use dump->restore, or having to run all of the migrations.

**Comment by msullivan** [comment]
It would probably have to work by doing an edgedb dump/restore, but that's maybe fine

**Comment by scotttrinh** [comment]
@msullivan 

> It would probably have to work by doing an edgedb dump/restore, but that's maybe fine

Yeah it seems like if we expose a way to do this in the CLI, we can start with:

1. Create a temporary schema-only branch
2. Dump schema-only branch
3. Restore schema-only branch to target instance

And then if we come up with a better way to do this in the future, we can swap that in when possible, but use the same command.

**Comment by CallumAlder** [comment]
This would be superb
